src/app/frontend/replicasetdetail/replicasetdetail_controller.js

```diff
@@ -156,6 +165,19 @@ export default class ReplicaSetDetailController {
    */
 
   /**
+   * Updates replicas count in replica set
+   * @param {number} replicasCount
+   * @param {function(!backendApi.ReplicaSetDetail)=} opt_callback
+   * @param {function(!angular.$http.Response)=} opt_errback
+   * @private
+   */
+  updateReplicas_(replicasCount, opt_callback, opt_errback) {
+    this.replicaSetDetail.podsDesired = replicasCount;
```

Comment:
I'd go with creating a new struct specialized only for this task. This is very cheap in our current architecture to create a new custom endpoint, so go for it. In return, you get clear, testable and maintainable API that can do only the thing it is designed to do.